chinglee-iot/FreeRTOS-Kernel
# Fix coverity violations for SMP diff part.

Fix coverity violations for SMP diff part.

Description
-----------

Fix coverity violations for SMP diff part.

Test Steps
-----------

1. Clone my forked [FreeRTOS repo](https://github.com/ActoryOu/FreeRTOS/tree/kernel-coverity).
1. cd FreeRTOS/Test/Coverity
1. Update the compiler configuration in Coverity
`cov-configure --force --compiler cc --comptype gcc -- -std=gnu90`
1. Build the (pseudo) application
`cov-build --emit-complementary-info --dir cov-out make multicore1`
or
`cov-build --emit-complementary-info --dir cov-out make multicore2`
1. Go to the Coverity output directory (cov-out) and begin Coverity static analysis
`cd cov-out/`
`cov-analyze --dir . --coding-standard-config ../../FreeRTOS/Test/Coverity/coverity_misra.config`

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
